test/teiserver/protocols/spring/spring_auth_test.exs

```diff
@@ -497,14 +508,15 @@ CLIENTS test_room #{user.name}\n"
     _ = _recv_raw(socket)
 
     # Now we get flood protection after the rename
-    _send_raw(
-      socket,
-      "LOGIN #{new_name} X03MO1qnZdYdgyfeuILPmQ== 0 * LuaLobby Chobby\t1993717506 0d04a635e200f308\tb sp\n"
-    )
+    expected_server_response = "DENIED Flood protection - Please wait 20 seconds and try again"
 
-    reply = _recv_until(socket)
-    [accepted | _remainder] = String.split(reply, "\n")
-    assert accepted == "DENIED Flood protection - Please wait 20 seconds and try again"
+    server_response =
+      Teiserver.Support.Tachyon.poll_until(
+        fn -> _send_login_attempt_and_parse_reply_msg_from_server_socket(socket, new_name) end,
+        &(&1 == expected_server_response)
+      )
```

Comment:
I am not the biggest fan of the style here. The helper function is a mouthful, it's used only once and it's declared very far from this call site.
Since you're already using anonymous function anyway, I think I'd rather inline everything:

```elixir
    Teiserver.Support.Tachyon.poll_until(
      fn ->
        _send_raw(
          socket,
          "LOGIN #{new_name} X03MO1qnZdYdgyfeuILPmQ== 0 * LuaLobby Chobby\t1993717506 0d04a635e200f308\tb sp\n"
        )

        reply = _recv_until(socket)
        [accepted | _remainder] = String.split(reply, "\n")
        accepted
      end,
      &(&1 == expected_server_response)
    )
```